File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a,b):
    return [i for i in range(a, b)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc,a,b ,s):
    if s=='s':
        return sum(inpFunc(a,b))
    else:
        return sum(inpFunc(a,b))/(b-a)




print(applicatorFunc(listFunc,1, 5 , 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc = [a..b] 

--Define applicatorFunc

applicatorFunc inpFunc a b s | s = 's' =sum (inpFunc a b )
                             |othwerwise = (sum (inpFunc a b) )/5  
--applicatorFunc inpFunc a b s = 
    {-if s=='s' 
        then sum (inpFunc a b )
        else (sum (inpFunc a b) )/5  
-}
 
main = do
    let result = applicatorFunc inpFunc 1 5 'a' --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show(result))


